packages/node-firebird-driver/src/lib/impl/attachment.ts

```diff
@@ -100,11 +100,33 @@ export abstract class AbstractAttachment implements Attachment {
 				prepareOptions?: PrepareOptions,
 				executeOptions?: ExecuteOptions
 			}): Promise<Array<any>> {
+		return (await this.executeReturningAs<Array<any>>(transaction, sqlStmt, parameters, {
+			...options, json: false
+		})).row;
+	}
+
+	async executeReturningAs<T>(transaction: AbstractTransaction, sqlStmt: string, parameters?: Array<any>,
+						   options?: {
+							   prepareOptions?: PrepareOptions,
+							   executeOptions?: ExecuteOptions,
+							   json?: boolean
```

Comment:
I do not like mixing array/object functions, it's not TypeScript friendly and is confusing. I prefer to have separate functions: `executeReturning` / `executeReturningObject`, `fetch` / `fetchObject`.

Also, there is no `json` here. JSON is about serialization (strings) and we do not do that here.

Reply:
`fetchAs` uses typescript generics, which means the end user has the power to instruct the compiler with their own interfaces. 

`.fetch()` in the order hand is stricter since you know you'll always expect it to return a array of `Array<any>`.

`fetchObject` could be added to the API to work as `fetch()` but for objects. 

Both, `fetch` and  `fetchObject` should call `fetchAs` internally (No reason to duplicate code). 

`fetchAs` is like a lower level function for `fetch` and `fetchObject` and can be their for end users that want to obtain column labels and resultSet in a single call. 


```
export interface User {
  ID: number;
  NAME: string;
}


let output = .....fetchObject();
console.log(output)  => [{ID: 1, NAME: 'John'}, .....];


let output = .....fetch();
console.log(output)  => [ [ 1, 'John' ] ];



let output = .....fetchAs<[ number: string; ]>>({json: false})
console.log(output)  => { columns: ['ID', 'NAME'], results: [  [1, 'John'] ] };
// Typescript will be provide intellisense for this: 
// output.results[0][0]  <-- number

let output = .... fetchAs<User>({json: true});
console.log(output)  => { columns: ['ID', 'NAME'], results: [  [ { ID: 1, NAME: 'John' }  ] ] };
// Typescript will be provide intellisense for this: 
// output.results[0].? (ID and NAME) will be suggested by typescript intellisense. 

```

In what concerns the use of `json` to represent javascript Objects, I see no problem even knowing it doesn't really reflect the technical term, anyway, I must agree that as `asObject` is more a suitable term. What do you think about it?

Reply:
I think `fetchAs` could exist as internal function (in `AbstractResultSet` or not) and `json` parameter could be renamed to `asObject`. I'm against `fetchAs` as public function as it does not add value (one could choose between `fetch` and `fetchObject` already).

Reply:
I see no reason to omit something that may serve a real purpose. 
Having `fetchAs` as an options makes the API richer. 
On my latest commits I supply fetchObject, fetch and fetchAs. Same for executeReturning, executeReturningObject, executeReturningAs.

`json` property is now named `asObject`.
Tests for all options were included.